Lib/json/tool.py

```diff
@@ -25,24 +25,37 @@ def main():
                         help='a JSON file to be validated or pretty-printed')
     parser.add_argument('outfile', nargs='?', type=argparse.FileType('w'),
                         help='write the output of infile to outfile')
+    parser.add_argument('--no-ensure-ascii', action='store_true', default=False,
+                        help='Do not set ensure_ascii to escape non-ASCII characters')
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('--indent', default=4, type=int,
+                       help='Indent level for pretty-printing.')
+    group.add_argument('--no-indent', action='store_true', default=False,
```

Comment:
Nice idea. Implemented in 0f38c18178cb6e9cbbaa495bf5a7d879e93dc117.

As an aside, `argparse` in Python 3.6 seems to [have a bug](https://bugs.python.org/issue18943) where mutual exclusivity is not checked if you don't change the default value. So for example:

```
# Error
echo "[1,2,3]" | python Lib/json/tool.py --indent=5 --no-indent

# No error, compact mode
echo "[1,2,3]" | python Lib/json/tool.py --indent=4 --no-indent

# No error, indented
echo "[1,2,3]" | python Lib/json/tool.py --no-indent --indent=4
```

Something to keep in mind... assuming this will be fixed.